be/src/olap/rowset/segment_v2/inverted_index_reader.cpp

```diff
@@ -252,11 +252,12 @@ Status FullTextIndexReader::new_iterator(OlapReaderStatistics* stats, RuntimeSta
 
 Status FullTextIndexReader::query(OlapReaderStatistics* stats, RuntimeState* runtime_state,
                                   const std::string& column_name, const void* query_value,
-                                  InvertedIndexQueryType query_type,
+                                  PrimitiveType primitiveType, InvertedIndexQueryType query_type,
                                   std::shared_ptr<roaring::Roaring>& bit_map) {
     SCOPED_RAW_TIMER(&stats->inverted_index_query_timer);
-
-    std::string search_str = reinterpret_cast<const StringRef*>(query_value)->to_string();
+    auto&& storage_value =
```

Comment:
The duplicate code can be moved to `InvertedIndexIterator::read_from_inverted_index`